app/src/utils/spatial-utils.tsx

```diff
@@ -178,23 +84,78 @@ export const parseSpatialDataByType = (
   return { markerLayers: [occurrencesMarkerLayer], staticLayers: [boundaryStaticLayer] };
 };
 
-// checks which key should be used to identify the layer
-export const getFeatureLayerKey = (spatialRecord: ISpatialData, feature: Feature): string => {
-  let key = '';
-
-  if (isOccurrenceFeature(feature)) {
-    key = `${spatialRecord.associated_taxa}`;
+const occurrenceMarkerSetup = (
```

Comment:
Can you add some comments in/around this function.  

Its got some relatively complex chained calls, would be good to add some helper comments to guide a future dev through its purpose. 

Same goes for the two functions below as well.

In general for any functions like these, don't worry about over explaining things, even using examples when appropriate, comments are free and I'd rather we over do it than under do it.

Reply:
Especially if there is any business logic involved, that a future dev might not be aware of, and which can't be fully interpreted from the code.